File: Server/Hero.Server.Core/Models/Character.cs
namespace Hero.Server.Core.Models
{
    public class Character
    {
        public Guid Id { get; set; }
        public string UserId { get; set; }
        public Guid GroupId { get; set; }
        public string? IconUrl { get; set; }
        public int? Age { get; set; }
        public string? Inventory { get; set; }
        public string? Religion { get; set; }
        public string? Relationship { get; set; }
        public string? Notes { get; set; }
        public string? Profession { get; set; }
        public List<Skilltree> Skilltrees { get; set; }
        public string Name { get; set; }
        public string? Description { get; set; }
        public Race Race { get; set; }
        public Guid? RaceId { get; set; }

        public bool IsPublic { get; set; }
        public bool? ShareSkilltree { get; set; }
        public bool? ShareNotes { get; set; }
        public bool? ShareInventory { get; set; }
        public bool? ShareAbilities { get; set; }
        public bool? ShareAttributes { get; set; }

        public void Update(Character character)
        {
            this.Name = character.Name;
            this.Description = character.Description;
            this.Skilltrees = character.Skilltrees;
            this.RaceId = character.RaceId;
            this.Age = character.Age;
            this.Inventory = character.Inventory;
            this.Religion = character.Religion;
            this.Relationship = character.Relationship;
            this.Notes = character.Notes;
            this.Profession = character.Profession;
            this.IconUrl = character.IconUrl;
        }
    }
}

File: Client/lib/src/features/characters/domain/character.dart
import 'package:freezed_annotation/freezed_annotation.dart';

import '../../admin/skilltrees/domain/skilltree_overview.dart';
import '../../traits/domain/ability.dart';
import '../../traits/domain/attribute_value.dart';
import '../../traits/domain/race.dart';

part 'character.freezed.dart';
part 'character.g.dart';

@freezed
class Character with _$Character {
  const factory Character({
    required String id,
    required String name,
    int? age,
    String? iconUrl,
    String? description,
    String? profession,
    String? religion,
    String? relationship,
    String? notes,
    String? inventory,
    String? userId,
    @Default(false) bool isPublic,
    bool? shareSkilltree,
    bool? shareNotes,
    bool? shareInventory,
    bool? shareAbilities,
    bool? shareAttributes,
    @Default([]) List<AttributeValue> attributes,
    required String raceId,
    required Race race,
    @Default([]) List<SkilltreeOverview> skilltrees,
    @Default([]) List<Ability> unlockedAbilities,
  }) = _Character;

  factory Character.fromJson(Map<String, dynamic> json) => _$CharacterFromJson(json);
}
